examples/CurveSimulator.ts

```diff
@@ -176,4 +176,6 @@ export default class CurveSim implements IClientSimulatorImpl {
             },
         };
     }
+
+    updateSimulationState(data: Record<string, unknown>) {}
```

Comment:
I went ahead and added these missing implementations to make type checker happy.